project/src/main/java/com/google/sps/data/PlacesFetcher.java

```diff
@@ -68,10 +69,16 @@
     /**
      * PlacesFetcher constructor.
      *
-     * @param geoApiContext the GeoApiContext used for all Google GEO API requests
+     * @param textSearchRequestGenerator
+     *     used for generating the TextSearchRequests sent to Google Places API
+     * @param placeDetailsRequestGenerator
+     *     used for generating the PlaceDetailsRequests sent to Google Places API
      */
-    public PlacesFetcher(GeoApiContext geoApiContext) {
-        this.context = geoApiContext;
+    public PlacesFetcher(
```

Comment:
I think that you should accept those in the @VisibleForTesting c'tor only. 
Then you have one c'tor same as you had before, and one test c'tor that provides you the fake generators. 
The non-test c'tor creates the generators itself, simplifying the calling code (QueryServlet does not need to know how SearchRequestGenerator and PlaceDetailsGenerator are). 
WDYT?

Reply:
Very good idea. It bothered me too that the queryServlet is exposed to them.